### ISSUE TITLE
Cleanup IDisposable aspects

### ISSUE THREAD

**Issue by pamidur** [pamidur/aspect-injector#1]

**Comment by pamidur** [comment]
We need to discuss it. It could be done if target is disposable and aspect scope is instance. Right?

**Comment by YuriyIvon** [comment]
Right, I would even disable setting scope "Class" for aspects with IDisposable interface. Apart from that, we will have to inject IDisposable interface and its implementation automatically to targets not having it. If Dispose() already exists, it may happen that calls to aspect methods were injected into it, so we have to do the cleanup only after those methods were called.
